robot-logic/src/main/java/logic/controller/MainController.java

```diff
@@ -0,0 +1,109 @@
+package logic.controller;
+
+import logic.book.Book;
+import logic.book.Category;
+import logic.parser.EbooksComParser;
+import logic.parser.Parser;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+import static logic.book.Category.*;
+import static java.lang.Thread.sleep;
+
+/**
+ * This class is the main logic.controller of creeper. The main goal of the class is to create an object
+ * that iterates over all available kinds of parsers. Then, for each logic.parser type the thread (ParserClassThread)
+ * is created. The thread takes another mapping (which is a value inside fullMappings map) as an argument.
+ * The class also contains the list of available logic.parser classes (which is passed by a constructor).
+ */
+public final class MainController {
+
+    private static final int MAX_QUEUE_SIZE = 1000;
+    private final Map<Class<? extends Parser>, Map<Category, List<URIGenerator>>> fullMappings;
+    private final BlockingQueue<Book> bookQueue = new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
+
+    /**
+     * Default constructor.
+     */
+    public MainController() {
+        fullMappings = initialize();
+    }
+
+
+    /**
+     * Launches new logic.parser threads for all different types of parsers.
+     */
+    public void launch() throws InterruptedException {
+        ExecutorService mainExecutor = Executors.newFixedThreadPool(fullMappings.size());
+        ExecutorService queueExecutor = Executors.newSingleThreadExecutor();    // just single for now...
+        fullMappings.keySet().forEach(e -> mainExecutor.submit(new ParserClassThread(e, fullMappings.get(e), bookQueue)));
+        mainExecutor.shutdown();
+
+        sleep(1000);
```

Comment:
This one would really have an impact on performance. Also it is platform dependent and it will slow down tests for this method.